lib/server.js

```diff
@@ -24,6 +26,16 @@ const initServer = function(
     applications = apps.createAppsIndex(applications);
   }
 
+  // versions cached for efficient search when provided version is a range
+  this._cachedVersions = new Map();
+  applications.forEach((appVersions, appName) => {
+    const versions = Array.from(appVersions)
+      .filter(app => app[0] !== 'latest')
+      .map(version => [new semver.SemVer(version[0]), version[1]])
```

Comment:
It doesn't help much. The file is read top to bottom, and these variable names are somewhat misleading. Both `app` and `version` at lines 33 and 34 are actually arrays of `[version, app]`, AFAIU.